bootable-jar/WFCORE-5184_Check_Boot.adoc

```diff
@@ -0,0 +1,100 @@
+= Bootable JAR, check server and deployment state at boot time
+:author:           Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This feature is to enforce that a Bootable JAR that successfully booted is in a valid state (server and deployment started without error).
+During boot, when an invalid state is encountered, the Bootable JAR process is aborted.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5184[WFCORE-5184]
+
+=== Related Issues
+
+* 
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* Bootable JAR runtime
+
+=== Other Interested Projects
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[Bootable JAR Maven plugin]
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [ ] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+The JAR is auto-checking the server and deployment states at boot time in order to abort the process in case of errors.
+
+=== Hard Requirements
+
+* The boot checks are enabled by default. This is a change of behavior that needs to be documented.
+* Introduce `--check-boot=[true|false]` in order to disable the checks.
```

Comment:
You mean a -Dorg.wildfly.bootable.check.boot=true|false? I am wandering if tha tis really required if we have a --check-boot argument?

Reply:
That could work. What I more meant is something that could be stored in the config so something like:
```
/system-property=org.wildfly.bootable.check.boot:add(value=false)
```

I'd have to look at the WIP, which I will :), to see if this would even work. Not critical really just an idea.

Reply:
Ah, I got it. I think that it would work, we could get this property prior to do the check.